File: JavaScript solutions/268. Missing Number.js
// approach : 1
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function (nums) {
  var sum = 0,
    n = nums.length;
  for (var i = 0; i < n; i++) {
    sum += nums[i];
  }
  return (n * (n + 1)) / 2 - sum;
};

//approach :2
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function (nums) {
  nums.sort((a, b) => a - b);
  for (var i = 0; i < nums.length; i++) {
    if (i != nums[i]) return i;
  }
  return nums.length;
};

//approach : 3
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function (nums) {
  const map = new Map();
  for (let i = 0; i <= nums.length; i++) {
    map.set(i, 1);
  }
  for (var i = 0; i < nums.length; i++) {
    map.set(nums[i], 2);
  }
  for (const [key, value] of map.entries()) {
    if (value === 1) return key;
  }
  return 0;
};

File: C++ solutions/268. Missing Number.cpp
#include<bits/stdc++.h>
using namespace std;

// solution
//approach : 1
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int sum=0,n=nums.size();
        for(int i=0;i<n;i++){
            sum+=nums[i];
        }
        return (n*(n+1)/2)-sum;
    }
};

//approach :2
class Solution {
public:
    int missingNumber(vector<int>& nums) {
       sort(nums.begin(),nums.end());
       for(int i=0;i<nums.size();i++){
           if(i!=nums[i]) return i;
       }
       return nums.size();
    }
};

//approach : 3
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        map<int,int>m;
        for(int i=0;i<=nums.size();i++){
            m[i]++;
        }
        for(int i=0;i<nums.size();i++){
            m[nums[i]]++;
        }
        for(auto k:m){
            if(k.second==1) return k.first;
        }
        return 0;
    }
};